src/types/chain.ts

```diff
@@ -33,7 +33,7 @@ export interface ChainOperation<T> {
   walletMustMatchChainID?: boolean;
 
   /** Validates the provided parameters match requirements */
-  validate(params: T, walletStore: WalletStore): boolean;
+  validate(params: T, walletStore: WalletStore): boolean | Promise<boolean>;
```

Comment:
We currently have this as a boolean for cosmos side txs, but for eth side we rely on async helpers